src/messages.c.jinja

```diff
@@ -0,0 +1,47 @@
+// NOTE: This file has been generated automatically by “{{script}}”.
+//       Do not edit manually!
+
+#include <stdlib.h>
+#include "messages.h"
+
+static const struct xkb_message_entry xkb_messages[] = {
+    {% for entry in entries %}
+    { {#--#}{{ entry.message_code_constant}}{#, "{{entry.id}}", "{{entry.description}}"#}},
+    {% endfor %}
+    {0{#, NULL, NULL #}}
+};
+
+const struct xkb_message_entry*
+xkb_message_get(xkb_message_code_t code)
+{
+{# Binary search seems overkill for now
+    if (code < XKB_LOG_MESSAGE_MIN_CODE || code > XKB_LOG_MESSAGE_MAX_CODE)
+        return NULL;
+
+    size_t first = 0;
+    size_t last = sizeof(xkb_messages) / sizeof(*xkb_messages) - 2;
+
+    /* binary search in xkb_messages */
+    while (last >= first) {
+        size_t mid = (first + last) / 2;
+        if (xkb_messages[mid].code < code)
+            first = mid + 1;
+        else if (xkb_messages[mid].code > code)
+            last = mid - 1;
+        else /* found it */
+            return &xkb_messages[mid];
+    }
+#}
+    if (code < XKB_LOG_MESSAGE_MIN_CODE || code > XKB_LOG_MESSAGE_MAX_CODE)
+        return NULL;
+
+    size_t idx_max = sizeof(xkb_messages) / sizeof(*xkb_messages) - 1;
+
+    for (size_t idx = 0; idx < idx_max; idx++) {
```

Comment:
```suggestion
    for (size_t idx = 0; idx < ARRAY_SIZE(xkb_messages) - 1; idx++) {
```

(drop the `-1` if you remove the `{0}` from `xkb_messages`)

Reply:
Done